internal/pkg/manifest/http.go

```diff
@@ -0,0 +1,115 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package manifest
+
+import (
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"gopkg.in/yaml.v3"
+)
+
+// RoutingRuleConfigOrBool holds advanced configuration for routing rule or a boolean switch.
+type RoutingRuleConfigOrBool struct {
+	RoutingRuleConfiguration
+	Enabled *bool
+}
+
+// Disabled returns true if the routing rule configuration is explicitly disabled.
+func (r *RoutingRuleConfigOrBool) Disabled() bool {
+	return r.Enabled != nil && !aws.BoolValue(r.Enabled)
+}
+
+// UnmarshalYAML implements the yaml(v3) interface. It allows https routing rule to be specified as a
+// bool or a struct alternately.
+func (r *RoutingRuleConfigOrBool) UnmarshalYAML(value *yaml.Node) error {
+	if err := value.Decode(&r.RoutingRuleConfiguration); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if !r.RoutingRuleConfiguration.isEmpty() {
+		// Unmarshalled successfully to r.RoutingRuleConfiguration, unset r.Enabled, and return.
+		r.Enabled = nil
+		return nil
+	}
+
+	if err := value.Decode(&r.Enabled); err != nil {
+		return errors.New(`cannot marshal "http" field into bool or map`)
+	}
+	return nil
+}
+
+// RoutingRuleConfiguration holds the path to route requests to the service.
+type RoutingRuleConfiguration struct {
+	Path                *string                 `yaml:"path"`
+	ProtocolVersion     *string                 `yaml:"version"`
+	HealthCheck         HealthCheckArgsOrString `yaml:"healthcheck"`
+	Stickiness          *bool                   `yaml:"stickiness"`
+	Alias               Alias                   `yaml:"alias"`
+	DeregistrationDelay *time.Duration          `yaml:"deregistration_delay"`
+	// TargetContainer is the container load balancer routes traffic to.
+	TargetContainer          *string `yaml:"target_container"`
+	TargetContainerCamelCase *string `yaml:"targetContainer"` // "targetContainerCamelCase" for backwards compatibility
+	AllowedSourceIps         []IPNet `yaml:"allowed_source_ips"`
+}
+
+// GetTargetContainer returns the correct target container value, if set.
+// Use this function instead of getting r.TargetContainer or TargetContainerCamelCase directly.
+func (r *RoutingRuleConfiguration) GetTargetContainer() *string {
```

Comment:
just to validate, is this the only method added? everything else was just a mv operation?

Reply:
yeah! and this function is effectively the same as the previous [`targetContainer()`](https://github.com/aws/copilot-cli/pull/3550/files#diff-8dc63309e718c405097ab0c6d168d527aa59e882bad9e0a0a144447c8e692a6dL258), just exported.